### PR TITLE
Add raytyping extension

### PR DESCRIPTION
**Pull request by louishuyng**
## Description
This extension inspired by monkeytype. For developer who loves typing keyboard and measure their speed of typing skill

## Screencast
SCR-20231203-lsoe
SCR-20231203-lsrl

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/raytyping/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "name": "raytyping",
+  "title": "Raytyping",
+  "description": "This extension inspired by monkeytype. For developer who loves typing keyboard and measure their speed of typing skill",
+  "icon": "command-icon.png",
+  "author": "louishuyng",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "raytype",
+      "title": "Raytyping",
```
Comment (pernielsentikaer) [inline extensions/raytyping/package.json:12]:
```suggestion
      "title": "Start Raytyping",
```

File: extensions/raytyping/src/raytype.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  return (
+    <List
+      searchText={typedText}
+      isLoading={inProgress}
+      isShowingDetail={true}
+      searchBarPlaceholder="Type here"
+      onSearchTextChange={(value) => setTypedText(value)}
+      searchBarAccessory={
+        <List.Dropdown
+          tooltip="Word amount"
```
Comment (pernielsentikaer) [inline extensions/raytyping/src/raytype.tsx:30]:
```suggestion
          tooltip="Word Amount"
```

File: extensions/raytyping/src/raytype.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    >
+      <List.Item
+        icon={{
+          source: Icon.Stopwatch,
+          tintColor: isCorrect ? Color.Green : Color.Red,
+        }}
+        title={`${speed} wpm`}
+        actions={
+          <ActionPanel>
+            <Action title="Reload" onAction={() => reloadGame()} />
```
Comment (pernielsentikaer) [inline extensions/raytyping/src/raytype.tsx:51]: Could add an icon here

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Missing [metadata images](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)

- I'm not sure about the loading indicator, it kinda steals the focus

- Is there a way to load real sentences instead of just words, it might be easier to type 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by louishuyng** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
>     * Missing [metadata images](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)
> 
>     * I'm not sure about the loading indicator, it kinda steals the focus
> 
>     * Is there a way to load real sentences instead of just words, it might be easier to type 🙂
> 
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

I'm not sure about the loading indicator, it kinda steals the focus
=> After testing and playing many times, I don't think it distracts us when typing. It also helps to indicate the typing now recording

Is there a way to load real sentences instead of just words, it might be easier to type 🙂
=> I want to keep monkey type experience, so I read their open source https://github.com/monkeytypegame/monkeytype  here and try to replicate the same experience as they did when typing

**Comment by louishuyng** [comment]
Hi @pernielsentikaer Do i need any update on this PR to make it releasable? It has been reviewed 2 weeks ago

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**